### PR TITLE
feat: `[Exiled::API]` StopDancing and StartDancing for SCP-3114 Role

### PR DESCRIPTION
**Pull request by NotZer0Two**
## Description
**Describe the changes** 
Adding the ability for Dancing with API

**What is the current behavior?** (You can also link to an open issue here)
The game by itself doesn't allow but there's left code from the client never used and so i took the variable and used it for the same reason

**What is the new behavior?** (if this is a feature change)
Ability to Dance

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:

<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [X] I have checked the project can be compiled
- [X] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [X] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Comment by NotZer0Two** [comment]
Don't know if i need to add IsDancing and other stuff to this let me know

**Comment by NotZer0Two** [comment]
And another issue while i was testing who made DanceType didn't match them

File: EXILED/Exiled.Events/Patches/Fixes/ServerSideDancesPatch.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label skip = generator.DefineLabel();
+
+            newInstructions.Add(new CodeInstruction(OpCodes.Ret));
+            newInstructions[newInstructions.Count - 1].labels.Add(skip);
+
+            newInstructions.InsertRange(7, new List<CodeInstruction>()
```
Comment (VALERA771) [inline EXILED/Exiled.Events/Patches/Fixes/ServerSideDancesPatch.cs:37]: Can you get index through FindIndex method? Setting it just like number is not safety iirc

File: EXILED/Exiled.Events/Patches/Fixes/ServerSideDancesPatch.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+            foreach (CodeInstruction instruction in newInstructions)
+                yield return instruction;
+
+            ListPool<CodeInstruction>.Pool.Return(newInstructions);
+        }
+
+        private static void Postfix(ref Scp3114Dance __instance, NetworkWriter writer)
+        {
+            Npc npc = Npc.Get(__instance.Owner);
```
Comment (VALERA771) [inline EXILED/Exiled.Events/Patches/Fixes/ServerSideDancesPatch.cs:50]: I think here should be player

**Review by VALERA771 — APPROVED** [review]
Test it pls

**Comment by VALERA771** [comment]
Here should be just comment, not approval

**Comment by NotZer0Two** [comment]
Ok so i did an error in the code that made everything strange, im now using an internal value that is inside the class but actually never used

**Comment by NotZer0Two** [comment]
And found out that if the player has a skin can dance but only for the client

**Review by VALERA771 — APPROVED** [review]

**Comment by VALERA771** [comment]
Also in future pls use for loop with `z` parameter. Its just some kind of tradition

**Comment by NotZer0Two** [comment]
> Also in future pls use for loop with `z` parameter. Its just some kind of tradition <- for transpilers

I can change it no problem with that

**Review by Misfiy — APPROVED** [review]

**Pull request merged**